### DISCUSSION TITLE
Implement a Queue System

### DISCUSSION THREAD

**Discussion by EncodedBird** [open-webui/open-webui#4256] [Ideas]
What steps, broadly, would be required to provide queue support in open-webui?

Basically since ollama can only handle one request at a time, rather than timing out it would be nice if it had a queue where it could tell you something like "The system is currently serving another request, you are {n} in the queue" (ideally customizable). Then it could automatically submit the request when it is the users turn rather than simply timing out because ollama is in use.

While ollama may implement parallel support one day, I don't see myself having enough resources to run multiple parallel models any time soon, so I feel the queue is a more practical solution.

You could think of this discussion as an extension from this one https://github.com/open-webui/open-webui/discussions/816

**Comment by Logan-Bolton** [comment]
also currently having this issue. Limited hardware, multiple users. If one query runs at a time its fine. But the moment more than one query gets asked at the same time, everything locks up and I have to reset ollama. I need some way to only let one query through at a time

**Reply by CryptITAustria** [reply]
Setting the olama environment configs will solve this problem, it does not seem to be an open-webui issue
https://www.llamafactory.cn/ollama-docs/en/faq.html#how-does-ollama-handle-concurrent-requests

```
- OLLAMA_MAX_LOADED_MODELS=1
- OLLAMA_NUM_PARALLEL=1
```
